db/seeds.rb

```diff
@@ -1,7 +1,65 @@
-# This file should contain all the record creation needed to seed the database with its default values.
-# The data can then be loaded with the bin/rails db:seed command (or created alongside the database with db:setup).
-#
-# Examples:
-#
-#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+require 'database_cleaner/active_record'
+require 'faker'
+
+puts "Clearing out the battlefield 🌋..."
+DatabaseCleaner.strategy = :truncation
+DatabaseCleaner.clean
+puts "Battlefield Cleared ⛰"
+
+puts "Summoning Gandalf 🧙🏻‍♂️"
+gandalf = User.create!(
+  email: 'gandalf@gmail.com',
+  nickname: 'GandalfTheWhite',
+  password: '123456',
+  city: 'Tokyo'
+  )
+puts "#{gandalf.nickname} Summoned: 'You Shall Not Pass! 🧖🏼‍♂️'"
+
+puts "Generating NPCs..."
+40.times do
+  user = User.new
+  user.email = Faker::Internet.email
+  user.nickname = Faker::Internet.username
+  user.password = '123456'
+  user.city = Faker::Address.city
+  user.save
+  puts "#{user.nickname} from #{user.city} created"
+end
+puts "All NPCs Generated"
+
+puts "Generating Groups..."
+20.times do
+  group = Group.new
+  group.name = Faker::App.name
+  group.description = Faker::Company.bs
+  group.city = Faker::Address.city
+  group.gm = User.randuser
+  group.save
+  puts "#{group.name} created"
+end
+puts "All Groups Created"
+
+puts "Linking Users to Groups..."
+30.times do
+  user = randUser
+  group = randGroup
+  group = randGroup if group.gm == user
+  p_group = PlayerGroup.new
+  p_group.user = user
+  p_group.group = group
+  p_group.save
+end
+puts "We have Player Groups 🕺"
+
+
+# See why this doesnt work...
+private
+
+def randUser
+  User.all.sample
+end
+
+def randGroup
+  Group.all.sample
+end
```

Comment:
ok, there's a lot wrong here...

1. follow ruby conventions for naming (i.e. `rand_user` not `randUser`)
2. there is no point putting it in a method if you are not reusing it
3. writing `private` only works withing the context of a Class or Module
4. you don't have a guarantee that the second time you call rand_group the user won't be a gm

look into either using an iterator, or doing a db query with conditions

Reply:
This part I was going to mention still in works and testing myself (this morning) shouldnt have added it to the commit.
Will follow your instructions